crates/sui-framework/sources/coin.move

```diff
@@ -119,7 +112,7 @@ module sui::coin {
         balance::join(balance, into_balance(coin));
     }
 
-    // === Functionality for Coin<T> holders ===
+    // === Base Coin functionality ===
```

Comment:
a naive q, for public functions related to split and join, why not have one function that takes a vector of coins and transforms them into a new list of coins with specified amounts, like what we do with `pay` in native [Sui transaction](https://github.com/gegaowp/sui/blob/602c1b266a9eea04ab3b06d1a5d4569b5c4ed195/crates/sui-types/src/messages.rs#L124)? 
In that sense, other modules can always count on this one function to do all all split and merging. The idea is very similar @oxade 's transformer change? https://github.com/MystenLabs/sui/pull/4528/files
cc @sblackshear

Reply:
I think this would be a very useful function to add this this new `wallet` module.

I do think that's it's separately useful to have `split`/`join` for singleton coins, only because it's somewhat cumbersome to put coins into vectors + take them out (and you will eventually have to take them out in order to transfer or spend them).

Reply:
Yes we plan to add this function in future, which is why I paused transformer.
Adding this requires a bit more thought (especially if we want to reuse the Pay API logic by putting common code in Rust/natives).
I'd rather do that as a separate more targeted PR.